### ISSUE TITLE
Check for consistent indentation and whitespace

### ISSUE THREAD

**Issue by jimmynotjim** [cfpb/cfgov-refresh#666]
@anselmbradford [pointed out L:313](https://github.com/cfpb/cfgov-refresh/pull/660#discussion_r32529145) in `_includes/post-macros.html` but I'm sure there's others. 

There are also a few instances of inconsistent whitespace [here](https://github.com/cfpb/cfgov-refresh/pull/660#discussion_r32544044) and [here](https://github.com/cfpb/cfgov-refresh/pull/660#discussion_r32544104)

It's probably a good time for a quick sweep to make sure all files are indented correctly and excess whitespace is removed.

**Comment by jimmynotjim** [comment]
Adam and I were discussing [Prettier](https://prettier.io/) today. It's opinionated, and may not match all our personal preferences, but being able to offload nonsense like this from each of our mental weights would be 💯. Has anyone else used it and have some first person feedback or recommendations?

TL;DR; is that you write the code however you want and on save Prettier reformats it to match _its_ standards. You can't customize from what I've seen, but I'm ok with that if I never have to note an indentation mixup in a PR ever again.

**Comment by anselmbradford** [comment]
Pretty sure we could use stylelint with the autofix flag to fix these too, and since we already use that elsewhere and it's customizable I'd prob favor it over prettier.

**Comment by jimmynotjim** [comment]
The benefit of Prettier is that it runs in your IDE of choice, so no one has to remember to run it, and the fixes are limited to formatting, it won't try and fix other non-stylistic linting issues. If we did use Prettier, we would still want to use Linters to catch larger issues like improper conditions or use of returns, etc. In the case of the Atom plugin, it comes with [ESLint integration as well](https://github.com/prettier/prettier-atom).

**Comment by anselmbradford** [comment]
Seems like it could be on a per-dev basis to use Prettier or not since it's local in the IDE, but the project should prob still have a matching stylelint config overall.

**Comment by Scotchester** [comment]
It looks to me like Prettier could be useful for our JS and CSS, _if_ we can get people to install the plugin in their editor. It all hinges on that. (It doesn't appear to support plain HTML or Jinja templates, from my initial look at it, but I'm OK with that, since my strongest opinions about formatting impacting code readability are reserved for HTML.)

Conceivably, we could cook up a script to run Prettier in Travis before merges, as a backstop for folks not having run Prettier before submitting the PR.

**Comment by ascott1** [comment]
@Scotchester Potentially of use: Prettier can also be used as a precommit hook, applying the formatting before a commit can be made, even if the user doesn't have the editor extension installed.

**Comment by Scotchester** [comment]
Yeah, that would be cool, too, though again, we'd have to get people to install it.

**Comment by jimmynotjim** [comment]
Some light research looks like we can install git hooks as part of `npm install`.

**Comment by anselmbradford** [comment]
I'd avoid pre-commit. I fear it'll just make commits slower and become an annoyance. If we go the git hook route, how about pre-push instead?

**Comment by jimmynotjim** [comment]
> how about pre-push instead

Won't that require a new commit because the file will have changed?

**Comment by anselmbradford** [comment]
> Won't that require a new commit because the file will have changed?

Yeah, my guess on how it would work is you would `git push` and if prettifier had things it needed to do it would fail the push and fix the changes, you'd then commit the changes and re-push. That seems reasonable to me. I'd rather have a commit for any autofixed changes so I could see what was happening. Pre-commit would be the same, but it would be mixed with whatever other work had been done manually. The commit could be rebased locally if you didn't want it in history.

**Comment by jimmynotjim** [comment]
> I'd rather have a commit for any autofixed changes so I could see what was happening

That's a really good point. My initial thought was "why not post-commit" but then we might have a billion Prettier commits for a single request. Pre-push seems like the best of all worlds, but will definitely take some getting used to. We'd also want to have some way to block the push if Prettier made a change, so the user was required to commit the formatting changes and then push again.

**Comment by jimmynotjim** [comment]
Whoops, I definitely did not mean to close this. Sorry

**Comment by jimmynotjim** [comment]
We should probably go ahead and close this. Maybe some day we can get Husky and Prettier to work with our machines but till then this seems impossible.
